### DISCUSSION TITLE
Correct Way to get a UIWindow in Uno for MacCatalyst from Window.Handle

### DISCUSSION THREAD

**Discussion by cesarchefinho** [unoplatform/uno#16061] [Q&A]
I am getting exception on         

UIWindow macWindow = ObjCRuntime.Runtime.GetNSObject<UIWindow>(this.Handle.Handle) ?? throw new Exception("foi mal");

The excetion is that it cannot cast <UIWindow> from nint
Tried to create a managed reference from an object that already has a managed reference (type: MainPage)

because I what to put Main Window in FullScreen

Can someone help ?

```csharp
public sealed partial class MainPage : Page
{
    private AppWindow _appWindow;  
    
    public MainPage()
    {
        this.InitializeComponent();

        UIWindow macWindow = ObjCRuntime.Runtime.GetNSObject<UIWindow>(this.Handle.Handle) ?? throw new Exception("foi mal");

        macWindow.Bounds = UIScreen.MainScreen.Bounds;

        UIViewController mainViewController = new ();

        mainViewController.ModalPresentationStyle = UIModalPresentationStyle.FullScreen;

        macWindow.RootViewController = mainViewController;
        
        macWindow.MakeKeyAndVisible();
```

**Comment by jeromelaban** [comment]
@MartinZikmund may have an answer!

**Comment by MartinZikmund** [comment]
Can you try using `ApplicationView.GetForCurrentView().TryEnterFullScreenMode()`

In the upcoming release you will be able to use `window.AppWindow.SetPresenter(AppWindowPresenterKind.FullScreen)`, this is however only available in the latest prerelease packages of Uno Platform
